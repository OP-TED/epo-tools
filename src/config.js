const EPO_LATEST = {
  owner: 'OP-TED',
  repo: 'ePO',
  localDirectory: 'assets/ePO/master',
  latest: true,
}

const EPO_3_1_0 = {
  owner: 'OP-TED',
  repo: 'ePO',
  tag: 'v3.1.0',
  localDirectory: `assets/ePO/v3.1.0`,
}

const COMMON_VOCABS = {
  localDirectory: 'assets/common-vocabularies',
}

const branchName = 'develop'
const UNDER_REVIEW = {
  owner: 'OP-TED',
  repo: 'ePO',
  localDirectory: `assets/ePO/${branchName}`,
  branch: branchName,
}

export { EPO_LATEST, EPO_3_1_0, UNDER_REVIEW, COMMON_VOCABS }
