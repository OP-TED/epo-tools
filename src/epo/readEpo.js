import { readFileSync } from 'fs'
import { bufferToJson } from '../conceptualModel/ea-to-json.js'
import { filterBy, startsWith } from '../conceptualModel/filter.js'

function getJson ({ databasePath }) {
  // const databasePath = `${localPath}/${conceptualModelPath}`
  const buffer = readFileSync(databasePath)
  return bufferToJson({ buffer })
}

const narrowToEpo = (g) => startsWith(g, 'epo')

function getEpoJson ({ databasePath }) {
  const json = getJson({ databasePath })
  return narrowToEpo(json)
}

export { getJson, getEpoJson }
