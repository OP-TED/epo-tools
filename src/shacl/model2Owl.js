import { applyTemplate } from './applyTemplate.js'
import { toTurtle } from './owlTemplate.js'

async function toOwl ({ nodes, edges }) {
  return applyTemplate({ nodes, edges }, toTurtle)
}

export { toOwl }
