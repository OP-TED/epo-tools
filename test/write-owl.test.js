import { expect } from 'expect'
import toMatchSnapshot from 'expect-mocha-snapshot'
import { describe, it } from 'mocha'
import { PRODUCTION } from '../src/config.js'
import { getEpoJson } from '../src/epo/readEpo.js'
import { toOwl } from '../src/shacl/model2Owl.js'

expect.extend({ toMatchSnapshot })

describe('write-owl', () => {

  const epoJson = getEpoJson(PRODUCTION)

  it(`generates owl`, async function () {
    const { dataset, turtle, errors } = await toOwl(epoJson)
    expect(turtle).toMatchSnapshot(this)
  })

  it(`generates owl, empty graph`, async function () {
    const { dataset, turtle, errors } = await toOwl({ nodes: [], edges: [] })
    expect(turtle).toMatchSnapshot(this)
  })

})

