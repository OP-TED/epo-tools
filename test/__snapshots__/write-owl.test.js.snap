// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-owl generates owl, empty graph 1`] = `
"@prefix schema: <http://schema.org/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .
@prefix cpov: <http://data.europa.eu/m8g/> .
@prefix a4g: <http://data.europa.eu/a4g/ontology#> .
@prefix sfrml: <http://data.europa.eu/a4g/mapping/sf-rml/> .
@prefix a4g_shape: <http://data.europa.eu/a4g/data-shape#> .
@prefix eli: <http://data.europa.eu/eli/ontology#> .
@prefix m8g: <http://data.europa.eu/m8g/> .
@prefix time: <http://www.w3.org/2006/time#> .
@prefix person: <http://www.w3.org/ns/person#> .
@prefix locn: <http://www.w3.org/ns/locn#> .
@prefix shacl: <http://www.w3.org/ns/shacl#> .
@prefix org: <http://www.w3.org/ns/org#> .
@prefix adms: <http://www.w3.org/ns/adms#> .
@prefix rml: <http://semweb.mmlab.be/ns/rml#> .
@prefix r2rml: <http://www.w3.org/ns/r2rml#> .
@prefix at: <http://publications.europa.eu/ontology/authority/> .
@prefix atold: <http://publications.europa.eu/resource/authority/> .
"
`;
